test/box-luatest/gh_9247_bsize_on_concurrent_dml_and_alter_test.lua

```diff
@@ -0,0 +1,95 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group('gh-9247', {{rollback = false}, -- test box.commit()
+                              {rollback = true}}) -- test box.rollback()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.after_each(function(cg)
+    cg.server:exec(function()
+        box.space.test:drop()
+    end)
+end)
+
+-- Check that space:bsize() is correct for a built-in-background primary index
+-- with concurrent inserts.
+-- Based on `test/box/alter-primary-index-tuple-leak-long.test.lua'.
+g.test_concurrent_replace_and_alter = function(cg)
+    cg.server:exec(function(rollback)
+        local fiber = require('fiber')
+        local s = box.schema.space.create('test')
+        local idx = s:create_index('pk')
+        local alter_started = false
+        local bsize_after_replace = 0
+        local data = string.rep('a', 66)
+
+        -- Create a table with enough tuples to make primary index alter in
+        -- background.
+        for i = 0, 7000 do
+            s:insert{i, i, data}
```

Comment:
You should insert data in batches (say 100 tuples in one transaction). This will speed up the test.

Reply:
Done.